param/__init__.py

```diff
@@ -1483,3 +1494,117 @@ def get_range(self):
         return abbreviate_paths(self.path,super(MultiFileSelector, self).get_range())
 
 
+class Date(Number):
+    """
+    Date parameter of datetime type.
+    """
+
+    def __init__(self, default=None, allow_None=False, **kwargs):
+        allow_None = True if default is None else allow_None
+        super(Date, self).__init__(default=default, allow_None=allow_None, **kwargs)
+        self._check_value(default)
```

Comment:
I just checked, and I think it's all covered, except that in the superclass, _check_value is only run if the value isn't callable.  Avoiding that check is what allows the values to be dynamic, so I *think* deleting the constructor is the right thing to do here.